tests/ui/pages/desktop/base.py

```diff
@@ -26,6 +26,15 @@ def login(self, email, password):
     def logout(self):
         self.header.click_logout()
 
+    def wait_for_page_to_load(self):
+        body = self.find_element(By.TAG_NAME, 'body')
+        self.wait.until(lambda s: 'firefox' in body.get_attribute('class'))
+
+    @property
+    def wait_for_page(self):
+        url = self.selenium.current_url
+        self.wait.until(lambda _: self.selenium.current_url != url)
```

Comment:
The URL changing does not mean the page has finished loading. What you're waiting for depends on what your recent interaction was. If you caused a new page to load, then a `page.wait_for_page_to_load()` is appropriate, and then that method will need to be implemented. Maybe that method will wait for a certain URL, or maybe wait for an element to be displayed on the page. If you can provide some more details about the interaction and what happens immediately after it, I can help with some specifics and perhaps suggest a good explicit wait.